src/hooks/use-login-mutation.ts

```diff
@@ -0,0 +1,23 @@
+import { useMutation, gql } from "@apollo/client";
+import { Alert } from "react-native";
+import { storage } from "../../App";
+
+const LOGIN_MUTATION = gql`
+  mutation MakeLogin($auth: LoginInput!) {
+    login(data: $auth) {
+      token
+    }
+  }
+`;
+
+export function useLoginMutation(email: string, password: string) {
+  const [loginMutation, { data, loading, error }] = useMutation(
+    LOGIN_MUTATION,
+    {
+      variables: { auth: { email: email, password: password } },
```

Comment:
normalmente quando utilizamos mutations a gente costuma passar as variáveis diretamente na chamada da mutation, mais ou menos isso:
```
      loginMutation({variables: { auth: { email: email, password: password } })
```
Até pra facilitar e evitar do componente que for utilizar a mutation tenha que saber qual o mapeamento de variáveis a gente faz um outra função que chama a mutate:
```
login = (email: string, password: string) => loginMutation({variables: { auth: { email, password } })
```

outro ponto é que quando temos variáveis com o mesmo nome da propriedade do objeto mapeado a gente pode passar ela direto:
```suggestion
      variables: { auth: { email, password } },
```